chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -1,15 +1,106 @@
 #[cfg(feature = "evm")]
-use chain_evm::state::{AccountTrie, Balance};
+use super::Error;
+#[cfg(feature = "evm")]
+use crate::smartcontract::Contract;
+#[cfg(feature = "evm")]
+use chain_evm::{
+    machine::{Config, Environment, Log, VirtualMachine},
+    state::{AccountTrie, Balance},
+};
 
 #[derive(Default, Clone, PartialEq, Eq)]
 pub struct Ledger {
     #[cfg(feature = "evm")]
     pub(crate) accounts: AccountTrie,
+    #[cfg(feature = "evm")]
+    pub(crate) logs: Vec<Log>,
```

Comment:
This is not very fork friendly, I think we should use some persistent data structure (even a dummy HAMT with indexes as keys).